torch/csrc/autograd/variable.cpp

```diff
@@ -137,26 +125,15 @@ std::shared_ptr<Function>& VariableViewImpl::get_grad_fn() {
   return _grad_fn;
 }
 
-void VariableViewImpl::rebase_history(VarFlags flags, int output_nr, std::shared_ptr<Function> grad_fn) {
+void VariableViewImpl::rebase_history(int output_nr, std::shared_ptr<Function> grad_fn) {
   TORCH_ASSERT(output_nr == 0);
-  TORCH_ASSERT(flags.requires_grad == bool(grad_fn));
-  if (grad_fn) {
-    TORCH_ASSERTM(grad_fn->num_inputs == 1, "Functions which modify views in-place must return a single Variable");
-  } else {
-    // TODO: perhaps we should enable this case by setting base.requires_grad=False
-    // and base.grad_fn = nullptr.
-    TORCH_ASSERTM(!base.requires_grad(), "base.requires_grad does not match view.requires_grad");
-  }
-  this->requires_grad = flags.requires_grad;
-  this->is_volatile = flags.is_volatile;
+  TORCH_ASSERT(grad_fn);
+  TORCH_ASSERTM(grad_fn->num_inputs == 1, "Functions which modify views in-place must return a single Variable");
   this->output_nr = output_nr;
-  base.requires_grad() |= flags.requires_grad;
-  base.is_volatile() |= flags.is_volatile;
-  if (grad_fn) {
-    base.output_nr() = 0;
-    base.get()->_grad_fn = std::make_shared<CopySlices>(
-        base, TensorGeometry(data), std::move(grad_fn));
-  }
+  base.output_nr() = 0;
+  base.get()->_grad_fn = std::make_shared<CopySlices>(
+      base, TensorGeometry(data), std::move(grad_fn));
+  get_grad_fn();  // trigger an update to the view's grad_fn
```

Comment:
There's one non-documented invariant - setting the flags needs to happen **after** we increment the version counter, or we won't trigger an update here. Would be nice to mention this next to the template in the code gen.

Reply:
See #4272

Reply:
You won't get the immediate update to `grad_fn` but it will still be correct if you swap the order. The next time `grad_fn()` is accessed it will update the function.